service/history/historyReplicator.go

```diff
@@ -947,14 +947,17 @@ func (r *historyReplicator) terminateWorkflow(
 					domainID,
 					workflowID,
 					runID,
-					msBuilder.GetNextEventID(),
+					mutableState.GetNextEventID(),
 				)
 			}
 
 			// setting the current version to be the last write version
-			msBuilder.UpdateReplicationStateVersion(currentLastWriteVersion, true)
```

Comment:
all `UpdateReplicationStateVersion` in this file may need to be changed to `UpdateCurrentVersion`

Reply:
replaced